File: 2446-determine-if-two-events-have-conflict/2446-determine-if-two-events-have-conflict.py
class Solution:
    def haveConflict(self, a: List[str], b: List[str]) -> bool:
        def g(s):
            return int(s[:2]) * 60 + int(s[3:])
        a[0] = g(a[0])
        a[1] = g(a[1])
        b[0] = g(b[0])
        b[1] = g(b[1])
        l = max(a[0], b[0])
        r = min(a[1], b[1])
        return l <= r
        

File: 2446-determine-if-two-events-have-conflict/2446-determine-if-two-events-have-conflict.cpp
class Solution {
public:
    
    int g(string & str) {
        return stoi(str.substr(0, 2)) * 60 + stoi(str.substr(3, 2));
    }
    
    bool haveConflict(vector<string>& e1, vector<string>& e2) {
        return max(g(e1[0]), g(e2[0])) <= min(g(e1[1]), g(e2[1]));
    }
};